### ISSUE TITLE
Custom gradient in `interfaces.keras.QuantumModel` sometimes not defined

### ISSUE THREAD

**Issue by renatomello** [qiboteam/qiboml#179]
When calling the method `interfaces.keras.QuantumModel.call`, there are a few options: (_i_) if `self.differentiation is `None`, some custom update of the parameters is returned; (_ii_) if `self.differentiation is not None` but `self.differentiation._is_built` is `False`, then `self.differentiation` is built and `self.custom_gradient` is defined. However, there is no definition of `self.custom_gradient` if `self.differentiation is not None` **and** `self.differentiation._is_built is True`. Therefore, when `self.custom_gradient.evaluate` is called, it raises an error because `self.custom_gradient is None`.
